src/IECoreScene/bindings/CurvesAlgoBinding.cpp

```diff
@@ -48,19 +48,42 @@ using namespace IECoreScene;
 namespace
 {
 
-boost::python::list segment(const CurvesPrimitive *curves, const PrimitiveVariable &primitiveVariable, const IECore::Data *segmentValues = nullptr )
+void resamplePrimitiveVariableWithRelease( const CurvesPrimitive *curvesPrimitive, PrimitiveVariable &primitiveVariable, PrimitiveVariable::Interpolation interpolation, const IECore::Canceller *canceller )
 {
-	boost::python::list returnList;
+	IECorePython::ScopedGILRelease gilRelease;
+	CurvesAlgo::resamplePrimitiveVariable( curvesPrimitive, primitiveVariable, interpolation, canceller );
+}
+
+CurvesPrimitivePtr deleteCurvesWithRelease( const CurvesPrimitive *curvesPrimitive, const PrimitiveVariable &curvesToDelete, bool invert, const IECore::Canceller *canceller )
+{
+	IECorePython::ScopedGILRelease gilRelease;
+	return CurvesAlgo::deleteCurves( curvesPrimitive, curvesToDelete, invert, canceller );
+}
+
+
+boost::python::list segment(const CurvesPrimitive *curves, const PrimitiveVariable &primitiveVariable, const IECore::Data *segmentValues = nullptr, const Canceller *canceller = nullptr )
```

Comment:
This `*WithRelease` naming seems a bit awkward when some wrapper functions do other work too - with `segment()` being an obvious example. We don't really have consistent naming for any of this, but of late I've been settling towards `*Wrapper`.

Reply:
Made everything consistently use Wrapper.